spec/forms/hub/take_action_form_spec.rb

```diff
@@ -93,4 +197,162 @@
+      let(:tax_return) { create :tax_return, client: client, year: 2019 }
+
+      context "only status change" do
+        let(:form) { Hub::TakeActionForm.new(client, current_user, { tax_return_id: tax_return.id, status: "intake_more_info", message_body: "" }) }
+
+        it "changes the status of the client" do
+          expect {
+            form.take_action!
+            tax_return.reload
+          }.to change(tax_return, :status).to "intake_more_info"
+        end
+
+        it "does not send an email" do
+          expect {
+            form.take_action!
+          }.not_to change(OutgoingEmail, :count)
+        end
+
+        it "does not send a text" do
+          expect {
+            form.take_action!
+          }.not_to change(OutgoingTextMessage, :count)
+        end
+
+        it "does not create a note" do
+          expect {
+            form.take_action!
+          }.not_to change(Note, :count)
+        end
+
+        it "creates an action list that only includes changing status" do
+          form.take_action!
+          expect(form.action_list).to eq ["updated status"]
+        end
+      end
+
+      context "status change with default message" do
+        let(:form) { Hub::TakeActionForm.new(client, current_user, { tax_return_id: tax_return.id, status: "intake_more_info" }) }
+        it "changes the status of the client" do
+          expect(SystemNote).to receive(:create_status_change_note).with(current_user, tax_return)
+
+          expect {
+            form.take_action!
+            tax_return.reload
+          }.to change(tax_return, :status).to "intake_more_info"
+        end
+
+        it "sends an email" do
+          expect {
+            form.take_action!
+          }.to change(OutgoingEmail, :count).by(1)
+        end
+
+        it "does not send a text" do
+          expect {
+            form.take_action!
+          }.not_to change(OutgoingTextMessage, :count)
+        end
+
+        context "when client has an sms preference" do
+          before do
+            allow(client.intake).to receive(:email_notification_opt_in_no?).and_return true
+            allow(client.intake).to receive(:sms_notification_opt_in_yes?).and_return true
+          end
+
+          it "sends an email" do
+            expect {
+              form.take_action!
+            }.to change(OutgoingEmail, :count).by(0)
+          end
+
+          it "does not send a text" do
+            expect {
+              form.take_action!
+            }.to change(OutgoingTextMessage, :count).by(1)
+          end
+        end
+
+        it "does not create a note" do
+          expect {
+            form.take_action!
+          }.not_to change(Note, :count)
+        end
+
+        it "creates an action list that only includes changing status" do
+          form.take_action!
+          expect(form.action_list).to eq ["updated status", "sent email"]
```

Comment:
The test says "only includes changing status" (L306) but the assertion includes "sent email" (L308). Is that an inconsistency?